frontend/packages/client/src/hooks/useVoteOnProposal.js

```diff
@@ -0,0 +1,70 @@
+import { useErrorHandlerContext } from 'contexts/ErrorHandler';
+import { useWebContext } from 'contexts/Web3';
+import { CAST_VOTE_TX } from 'const';
+import { checkResponse } from 'utils';
+import { useMutation } from '@tanstack/react-query';
+
+export default function useVoteOnProposal() {
+  const { notifyError } = useErrorHandlerContext();
+  const { user, signMessageByWalletProvider } = useWebContext();
+
+  const {
+    mutateAsync: voteOnProposal,
+    isLoading,
+    isError,
+    isSuccess,
+    data,
+    error,
+  } = useMutation(
+    async ({ proposal, voteData }) => {
+      const timestamp = Date.now();
+      const hexChoice = Buffer.from(voteData.choice).toString('hex');
+      const message = `${proposal.id}:${hexChoice}:${timestamp}`;
+      const hexMessage = Buffer.from(message).toString('hex');
+
+      const [compositeSignatures, voucher] = await signMessageByWalletProvider(
```

Comment:
Can we move this into a function in our `api` directory?

Reply:
created voteOnProposalApiReq fn but I think as a general approach we could leave signing the request in the react-query definition, otherwise we need to pass to the function the `signMessageByWalletProvider ` and the `user` since they come from a hook they can't be included in a regular function because of hook restrictions